iot/api-client/manager/cloudiot_device_manager_example.py

```diff
@@ -0,0 +1,330 @@
+#!/usr/bin/env python
+
+# Copyright 2017 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""Example of using the Google Cloud IoT Core device manager to administer
+devices.
+
+This example uses the Device Manager API to create, retrieve, disable, list and
+delete Cloud IoT Core devices and registries, using both RSA and eliptic curve
+keys for authentication.
+
+Before you run the sample, configure Cloud IoT Core as described in the
+documentation at https://cloud.google.com/iot or by following the instructions
+in the README located in the parent folder.
+
+Usage example:
+
+  $ python cloudiot_device_manager_example.py \
+      --project_id=my-project-id \
+      --pubsub_topic=projects/my-project-id/topics/my-topic-id \
+      --api_key=YOUR_API_KEY \
+      --ec_public_key_file=ec_public.pem \
+      --rsa_certificate_file=rsa_cert.pem \
+      --service_account_json=service_account.json
+
+Troubleshooting:
+
+  - If you get a 400 error when running the example, with the message "The API
+    Key and the authentication credential are from different projects" it means
+    that you are using the wrong API Key. Ensure that you are using the API key
+    from Google Cloud Platform's API Manager's Credentials page.
+"""
+
+import argparse
+import sys
+import time
+
+from googleapiclient import discovery
+from googleapiclient.errors import HttpError
+from oauth2client.service_account import ServiceAccountCredentials
+
+API_SCOPES = ['https://www.googleapis.com/auth/cloud-platform']
+API_VERSION = 'v1alpha1'
+DISCOVERY_API = 'https://cloudiot.googleapis.com/$discovery/rest'
+SERVICE_NAME = 'cloudiot'
+
+
+def discovery_url(api_key):
+    """Construct the discovery url for the given api key."""
+    return '{}?version={}&key={}'.format(DISCOVERY_API, API_VERSION, api_key)
+
+
+class DeviceRegistry(object):
```

Comment:
This class reads like something from a client library, not like something from a code sample. I would consider how you could reduce this class to stateless functions and how you could consider making the code flow through logical steps such as create registry, register a device, etc.